catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/SearchFormsApplication.java

```diff
@@ -112,17 +138,54 @@ public void init() {
                 .filter(Objects::nonNull)
                 .map(transformer::toFieldFilter)
                 .filter(Objects::nonNull)
+                .sorted(Comparator.comparing(CommonTemplate::getTitle))
                 .collect(Collectors.toList()),
         MAPPER::toJson);
 
+    put(
+        "/forms/query",
+        APPLICATION_JSON,
+        (req, res) ->
+            runWhenNotGuest(
+                res,
+                () ->
+                    doCreateOrUpdate(
+                        res,
+                        Stream.of(safeGetBody(req))
+                            .map(MAPPER::fromJson)
+                            .map(Map.class::cast)
```

Comment:
❓ Any reason we can't use the direct `parseMap` function so we don't have to cast?
```java
.map(MAPPER.parser()::parseMap)
```